test/e2e/tests/test_ca.py

```diff
@@ -0,0 +1,112 @@
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the ACMPCA Certificate resource.
+"""
+
+import time
+import logging
+import base64
+import pytest
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+from acktest.k8s.resource import _get_k8s_api_client
+from kubernetes import client
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_acmpca_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.tests.helper import ACMPCAValidator
+from e2e.fixtures import k8s_secret
+
+RESOURCE_PLURAL = "certificateauthorities"
+
+CREATE_WAIT_AFTER_SECONDS = 10
+UPDATE_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+@pytest.fixture(scope="module")
+def create_ca(acmpca_client):
+    ca_name = random_suffix_name("certificate-authority", 50)
+    replacements = {}
+    replacements["NAME"] = ca_name
+    replacements["COMMON_NAME"] = "www.example.com"
+    replacements["COUNTRY"] = "US"
+    replacements["LOCALITY"] = "Arlington"
+    replacements["ORG"] = "Example Organization"
+    replacements["STATE"] = "Virginia"
+
+    # Load CA CR
+    ca_resource_data = load_acmpca_resource(
+        "certificate_authority",
+        additional_replacements=replacements,
+    )
+
+    # Create k8s resource
+    ca_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+        ca_name, namespace="default",
+    )
+    k8s.create_custom_resource(ca_ref, ca_resource_data)
+    ca_cr = k8s.wait_resource_consumed_by_controller(ca_ref)
+
+    time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+    assert ca_cr is not None
+    assert k8s.get_resource_exists(ca_ref)
+    logging.info(ca_ref)
+
+    ca_resource_arn =  k8s.get_resource_arn(ca_cr)
+    assert ca_resource_arn is not None
+
+    yield (ca_cr, ca_resource_arn)
+
+    #Delete CA k8s resource
+    _, deleted = k8s.delete_custom_resource(ca_ref)
+    assert deleted is True
+
+    time.sleep(DELETE_WAIT_AFTER_SECONDS) 
+
+    # Check CA status is DELETED
+    acmpca_validator = ACMPCAValidator(acmpca_client)
+    acmpca_validator.assert_certificate_authority(ca_resource_arn, "DELETED")
+
+@service_marker
+class TestCertificateAuthority:
+
+    def test_create_delete(self, acmpca_client, create_ca):
+        
+        (ca_cr, ca_resource_arn) = create_ca
+        #ca_arn = ca_cr['status']['ackResourceMetadata']['arn']
+
+        # Check CA status is PENDING_CERTIFICATE
+        acmpca_validator = ACMPCAValidator(acmpca_client)
+        ca = acmpca_validator.assert_certificate_authority(ca_resource_arn, "PENDING_CERTIFICATE")
```

Comment:
Can we also check that the status in the kubernetes resource is "pending" as well?

Reply:
Yes, I'll add that in a separate PR since the CA resource doesn't have a status field, I'm planning to add it as a field in the CA resource's Status.